src/data/zcl_abapgit_data_deserializer.clas.abap

```diff
@@ -73,6 +73,33 @@ CLASS zcl_abapgit_data_deserializer IMPLEMENTATION.
   ENDMETHOD.
 
 
+  METHOD is_table_allowed_to_edit.
+
+    DATA lv_tabname TYPE dd02l-tabname.
+
+    "Did the user flagged this table for update?
+    READ TABLE is_checks-overwrite TRANSPORTING NO FIELDS
+      WITH KEY obj_name = iv_table_name decision = zif_abapgit_definitions=>c_yes.
+    IF sy-subrc <>  0.
+      RETURN.
+    ENDIF.
+
+    "For safety reasons only customer dependend customizing tables are allowed to update
+    SELECT SINGLE dd02l~tabname
+      FROM dd09l JOIN dd02l
+        ON dd09l~tabname = dd02l~tabname
+        AND dd09l~as4local = dd02l~as4local
+        AND dd09l~as4vers = dd02l~as4vers
+      INTO lv_tabname
+      WHERE dd09l~tabname = iv_table_name
+        AND dd09l~tabart = 'APPL2'
+        AND dd09l~as4user <> 'SAP'
+        AND dd09l~as4local = 'A' "Only active tables
+        AND dd02l~contflag = 'C'. "Only customizing tables
+    rv_allowed_to_edit = boolc( sy-subrc = 0 ).
```

Comment:
I need a user exit here, or some of the stuff I'm working on will not work anymore

Reply:
eg. see the breaking cross check in https://github.com/abapGit/abapGit/pull/6138/checks?check_run_id=12341891700

Reply:
agree. i will add an exit